core/arch/arm/plat-imx/sub.mk

```diff
@@ -8,6 +8,8 @@ ifeq ($(CFG_PSCI_ARM32),y)
 srcs-y += pm/psci.c pm/gpcv2.c
 srcs-$(CFG_MX7) += pm/pm-imx7.c pm/psci-suspend-imx7.S pm/imx7_suspend.c
 $(call force,CFG_PM_ARM32,y)
+asm-defines-file := $(sub-dir)/imx_pm_asm_defines.c
```

Comment:
This somewhat hackish.

Reply:
What is hackish exactly? The use of `$(sub-dir)` when other statements such as `srcs-y` use relative paths? Or the fact that simply including `mk/gen-asm-defines.mk` will generate a file? (would you rather recommend that we should call a function instead?)

Reply:
It's a bit of both, `asm-defines-y += imx_pm_asm_defines.c` without any special include would harmonize better with the rest.

However, I don't think it's much of an issue as we're likely to have at most one of these files per platform directory.

Reply:
I also prefer you suggestion. It makes things much cleaner. Patches updated and squashed.